src/VisualStudio/Core/Test.Next/Services/SolutionServiceTests.cs

```diff
@@ -488,27 +489,27 @@ private static async Task VerifySolutionUpdate(TestWorkspace workspace, Func<Sol
             var second = await service.GetSolutionAsync(newSolutionChecksum, CancellationToken.None);
 
             Assert.Equal(newSolutionChecksum, await second.State.GetChecksumAsync(CancellationToken.None));
-            Assert.False(object.ReferenceEquals(primaryWorkspace.PrimaryBranchId, second.BranchId));
+            Assert.NotSame(primaryWorkspace.PrimaryBranchId, second.BranchId);
 
             // do same once updating primary workspace
             await ((ISolutionController)service).UpdatePrimaryWorkspaceAsync(newSolutionChecksum, solution.WorkspaceVersion + 1, CancellationToken.None);
             var third = await service.GetSolutionAsync(newSolutionChecksum, CancellationToken.None);
 
             Assert.Equal(newSolutionChecksum, await third.State.GetChecksumAsync(CancellationToken.None));
-            Assert.True(object.ReferenceEquals(primaryWorkspace.PrimaryBranchId, third.BranchId));
+            Assert.Same(primaryWorkspace.PrimaryBranchId, third.BranchId);
         }
 
         private static async Task<SolutionService> GetSolutionServiceAsync(Solution solution, Dictionary<Checksum, object> map = null)
         {
             // make sure checksum is calculated
             await solution.State.GetChecksumAsync(CancellationToken.None);
 
-            map = map ?? new Dictionary<Checksum, object>();
+            map ??= new Dictionary<Checksum, object>();
             await solution.AppendAssetMapAsync(map, CancellationToken.None);
 
             var sessionId = 0;
             var storage = new AssetStorage();
-            var source = new TestAssetSource(storage, map);
+            _ = new TestAssetSource(storage, map);
```

Comment:
😕 Why have this?

Reply:
The instance registers itself with the storage in its constructor :-|

Reply:
ya. it was due to the circular reference code it has, you can break the circular dependencies if that is cleaner. 

this snapshot service that register static asset source and etc was due to the limitation - servicehub's static service pattern + service can only have 1 connection and all connection must be initiated from VS side.

which I end up solved by snapshot service.

but now servicehub - more strictly, StreamJsonRpc supports multiplex streams (https://github.com/AArnott/Nerdbank.Streams/blob/master/doc/MultiplexingStream.md) - supports VS creating initial connection to servicehub and then multiplex that stream to let servicehub create sub stream back to VS removing the need for snapshot service.

servicehub now fully support the multiplex stream and even added CreateAsync API for us so that we can create service asynchronously (rather than syncrhonous ctor), moving the multiplexing stream will let us remove SnapshotService and the above circular dependency.